src/test_sudoku.c

```diff
@@ -0,0 +1,134 @@
+ *
+ * Language : C
+ *
+ * Author   : Rishith Rao - rishithrao@cmail.carleton.ca
+ *            Prasanth Kota - prasanthkota@cmail.carleton.ca
+ *
+ * Date     : 13 June, 2020.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "../include/sudoku.h"
+
+struct node {
+    int data;
+    struct node *next;
+};
+
+struct node *head = NULL;
+struct node *current = NULL;
+
+/**
+ * Deletes the first element of the array.
+ */
+struct node* deleteFirst() {
+    struct node *tempLink = head;/*save reference to first link*/
+    head = head->next;/*mark next to first link as first*/
+    return tempLink;/*return the deleted link*/
+}
+
+/**
+ * Checks whether the list is empty or not.
+ */
+bool isEmpty() {
+    return head == NULL;
+}
+
+/**
+ * Insert link at the first location.
+ */
+void insertFirst(int data){
+    struct node *link = (struct node*) malloc(sizeof(struct node));/*create a link*/
+    link->data = data;
+    link->next = head;/*point it to old first node*/
+    head = link;/*point first to new first node*/
+}
+
+/**
+ * Delete a link with given data
+ */
+struct node* delete(int data) {
+    struct node* current = head;/*start from the first link*/
+    struct node* previous = NULL;
+    if(head == NULL){/*if list is empty*/
+        return NULL;
+    }
+    while(current->data != data){/*navigate through list*/
+        if(current->next == NULL) {/*if it is last node*/
+            return NULL;
+        }else {
+             previous = current;/*store reference to current link*/
+             current = current->next;/*move to next link*/
+         }
+    }
+    if(current == head){/*found a match, update the link*/
+        head = head->next;/*change first to point to next link*/
+    }else {
+        previous->next = current->next;/*bypass the current link*/
+     }    
+    return current;
+}
+
+/*
+ *This checks the 3X3 blocks for repeated values.
+ */
+_Bool test_3X3(int rearranged_sudoku[][9]) {
+    int z[9][9];  
+    int x=0,y=2,sum=0,k=0,l=2,a,sumplus=0;
+    int flag=0;
+    while(x<=8) {
+        while(k<=8) {
+            for (int i=x;i<=y;i++) {
+                for(int j=k;j<=l;j++) {
+   		    a=rearranged_sudoku[i][j];
+   		    sum=sum+a;
+                    delete(a);
+                }
+            }
+            k=k+3;
+            l=l+3;
+            sumplus=sum;
+            while(!isEmpty() || sumplus!=45) {/*checks whether the array is empty or not. If not it empties the array for furthur operation.*/
+                flag=true;  
+                sumplus=45;        
+                struct node *temp = deleteFirst();
+            }  
+            sum=0;
+    	    insertFirst(1);
+            insertFirst(2);
+            insertFirst(3);
+            insertFirst(4);
+            insertFirst(5);
+            insertFirst(6);
+            insertFirst(7);
+            insertFirst(8);
+            insertFirst(9);
```

Comment:
This entire portion is useless. It is not explain anywhere what this does. What I can see is that you are adding the numbers 1 through 9 into a linked list and then deleting them when you are doing your if() statement at the end of the block check. I will be frank. This verifies nothing. It does not verify anything in the rearanged_sudoku array and uses heap memory for nothing. This needs to go.

Reply:
There is a purpose to add the elements to linkedlists. This is also modified for better understanding for the other developers.